File: BlocksWorld_SPD/Unstack.cs
using it.unical.mat.embasp.languages;
using System;
using System.Collections.Generic;
using System.Text;

namespace BlocksWorld_SPD
{
    [Id("stack")]
    class Unstack
    {
        [Param(0)]
        private string block1;

        [Param(1)]
        private string block2;

        public Unstack()
        {
            this.block1 = "";
            this.block2 = "";
        }

        public Unstack(string block1, string block2)
        {
            this.block1 = block1;
            this.block2 = block2;
        }

        public string getBlock1()
        {
            return this.block1;
        }

        public string getBlock2()
        {
            return this.block2;
        }

        public void setBlock1(string block1)
        {
            this.block1 = block1;
        }

        public void setBlock2(String block2)
        {
            this.block2 = block2;
        }

        public override String ToString()
        {
            return "stack(block1 = " + block1 + ", block2 = " + block2 + ")";
        }
    }
}

File: EmbASP_Desktop_BlocksWorld_Python/src/Unstack.py
from languages.predicate import Predicate

class Unstack(Predicate):
    predicate_name = "stack"
    
    def __init__(self, block1=None, block2=None):
        Predicate.__init__(self, [("block1"),("block2")])
        self.block1 = block1
        self.block2 = block2
        
    def get_block1(self):
        return self.block1
    
    def get_block2(self):
        return self.block2
    
    def set_block1(self, block1):
        self.block1 = block1
        
    def set_block2(self, block2):
        self.block2 = block2
        
    def __str__(self):
        return "stack(block1 = " + self.block1 + ", block2 = " + self.block2 + ")"